src/test/java/com/ericsson/ei/notifications/EmailSenderTest.java

```diff
@@ -43,14 +44,29 @@
     @InjectMocks
     private EmailSender emailSender;
 
+    @Value("${email.sender:noreply@ericsson.com}")
+    private String sender;
+
+    @Value("${email.subject:Email notification}")
+    private String subject;
+
     @Test(expected = NotificationFailureException.class)
     public void sendEmailThrowsException() throws Exception {
         doThrow(new MailSendException("")).when(javaMailSender).send(message);
         emailSender.sendEmail(message);
     }
 
+    @Test
+    public void testDefaultValues() {
+    if (sender == null || subject == null) {
```

Comment:
in this case if one value is given but not the other both will be set to be default not just the missed one.

Reply:
Isn't it enough to extend these unit tests with more scenarios, to test? 1) only subject was given, 2) only sender was given 3) no configuration was given - default should be added. 
Then we don't have to extend the cucumber scenarios with several scenarios, but just one 😄

Reply:
Currently I don't understand what this tests. The values are fetched from the properties and they have a defined default. This means they cannot be null so the test would always fall to the else statement.